code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+	. = ..()
+	if(!iscyborg(affected_mob))
+		return
+	var/mob/living/silicon/borgo = affected_mob
+	borgo.adjustBruteLoss(-heal_amt)
+
+/obj/structure/slime_crystal/yellow
+	colour = "yellow"
+	light_color = LIGHT_COLOR_YELLOW //a good, sickly atmosphere
+	light_power = 0.75
+
+/obj/structure/slime_crystal/yellow/Initialize()
+	. = ..()
+	set_light(3)
+
+/obj/structure/slime_crystal/yellow/attacked_by(obj/item/I, mob/living/user)
+	if(istype(I,/obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/cell = I
+		//Punishment for greed
+		if(cell.charge == cell.maxcharge)
+			cell.explode()
+			return
+		cell.give(cell.maxcharge)
+		return
+	return ..()
+/obj/structure/slime_crystal/darkpurple
+	colour = "dark purple"
+
+/obj/structure/slime_crystal/darkpurple/process()
+	var/turf/T = get_turf(src)
+	if(!istype(T,/turf/open))
+		return
+	var/turf/open/open_turf = T
+	var/datum/gas_mixture/air = open_turf.return_air()
+
+	if(air.get_moles(/datum/gas/plasma) > 15)
+		air.adjust_moles(/datum/gas/plasma,-15)
+		new /obj/item/stack/sheet/mineral/plasma(open_turf)
+
+/obj/structure/slime_crystal/darkpurple/Destroy()
+	atmos_spawn_air("plasma=[20];TEMP=[500]")
+	return ..()
+
+/obj/structure/slime_crystal/darkblue
+	colour = "dark blue"
+
+/obj/structure/slime_crystal/darkblue/process()
+	var/list/listie = range(5,src)
+	for(var/turf/open/T in listie)
+		if(prob(75))
+			continue
+		var/turf/open/open_turf = T
+		open_turf.MakeDry(TURF_WET_LUBE)
+
+	for(var/obj/item/trash/trashie in listie)
+		if(prob(75))
+			continue
+		qdel(trashie)
+
+/obj/structure/slime_crystal/silver
+	colour = "silver"
+
+/obj/structure/slime_crystal/silver/process()
+	for(var/obj/machinery/hydroponics/hydr in range(5,src))
+		hydr.weedlevel = 0
+		hydr.pestlevel = 0
+		hydr.age++
+
+/obj/structure/slime_crystal/bluespace
+	colour = "bluespace"
+	density = FALSE
+	///Is it in use?
+	var/in_use = FALSE
+
+/obj/structure/slime_crystal/bluespace/Initialize()
+	. = ..()
+	GLOB.bluespace_slime_crystals += src
+
+/obj/structure/slime_crystal/bluespace/Destroy()
+	GLOB.bluespace_slime_crystals -= src
+	return ..()
+
+/obj/structure/slime_crystal/bluespace/attack_hand(mob/user)
+
+	if(in_use)
+		return
+
+	var/list/local_bs_list = GLOB.bluespace_slime_crystals.Copy()
+	local_bs_list -= src
+	if(local_bs_list.len == 0)
+		return ..()
+
+	if(local_bs_list.len == 1)
+		do_teleport(user ,local_bs_list[1])
+		return
+
+	in_use = TRUE
+
+	var/list/assoc_list = list()
+
+	for(var/BSC in local_bs_list)
+		var/area/bsc_area = get_area(BSC)
+		var/name = "[bsc_area.name] bluespace slimic pylon"
+		var/counter = 0
+
+		do
+			counter++
+		while(assoc_list["[name]([counter])"])
+
+		name += "([counter])"
+
+		assoc_list[name] = BSC
+
+	var/chosen_input = input(user,"What destination do you want to choose",null) as null|anything in assoc_list
+	in_use = FALSE
+
+	if(!chosen_input)
+		return
+
+	do_teleport(user ,assoc_list[chosen_input])
```

Comment:
What if the selected crystal is destroyed while the user has the input menu open?